client/src/components/newUser/newUser.jsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './newUser.sass';
+
+export class NewUser extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			username: '',
+			password: '',
+			repPassword: '',
+			firstName: '',
+			lastName: '',
+			passMismatch: false
+		}
+		this.handleUsernameChange = this.handleUsernameChange.bind(this);
+		this.handlePasswordChange = this.handlePasswordChange.bind(this);
+		this.handleRepPasswordChange = this.handleRepPasswordChange.bind(this);
+		this.handleFirstNameChange = this.handleFirstNameChange.bind(this);
+		this.handleLastNameChange = this.handleLastNameChange.bind(this);
+		this.showLoginPage = this.showLoginPage.bind(this);			
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.viewHomepage = this.viewHomepage.bind(this);
+	};
+
+	viewHomepage() {
+		this.props.return();
+	};
+
+	handleUsernameChange(event) {
+		this.setState({username: event.target.value});
+	};
+
+	handlePasswordChange(event) {
+		this.setState({password: event.target.value});
+	};
+
+	handleRepPasswordChange(event) {
+		this.setState({repPassword: event.target.value});
+	};
+
+	handleFirstNameChange(event) {
+		this.setState({firstName: event.target.value});
+	};
+	
+	handleLastNameChange(event) {
+		this.setState({lastName: event.target.value});
+	};
+
+	showLoginPage() {
+		this.props.viewLogin();
+	};
+
+	handleSubmit(event) {
+		event.preventDefault();
+
+		if (this.state.password === this.state.repPassword) {
+			this.submitUser();
+		} else {
+			this.setState({
+				passMismatch: true,
+				password: '',
+				repPassword: ''
+			});
+		};
+	};
+
+	submitUser() {
+		const newUser = {
+			username: this.state.username,
+			password: this.state.password,
+			firstName: this.state.firstName,
+			lastName: this.state.lastName,
+		};
+
+		const newUserInfo = {
+			method: 'post',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify(newUser)
+		};
+
+		fetch('/auth/v1/signup', newUserInfo)
+		.then((response) => response.json())
+		.then((response) => {
+			if (response.data) {
+				console.log(response);
+				const userData = response.data;
+				delete userData['password'];
+				console.log("Successfully created new user");
+
+				localStorage.setItem('user', JSON.stringify(userData));
+				this.setState({
+					username: '',
```

Comment:
This object is duplicated below, and is the "neutral" state right? If so it should be pulled out into a variable to be passed in here

Reply:
Yes, this would be the default 'fresh page' state.

So you'd like to have the neutral state be it's own variable declared outside of the if statement so that it's only written once?

Reply:
Yep. Then if we're changing how it defaults to, we only change it in one spot. Likely that wouldn't happen for a new user, but you never know